sdk/storage/inc/shares/protocol/share_rest_client.hpp

```diff
@@ -832,7 +833,7 @@ namespace Azure { namespace Storage { namespace Files { namespace Shares {
   {
     std::string ETag;
     std::string LastModified;
-    bool IsServerEncrypted = bool();
+    bool RequestIsServerEncrypted = bool();
```

Comment:
This variable name is kind of weird. .Net it's named `IsServerEncrypted`. In js, it's named `isServerEncrypted`. In Python, it's named `server_encrypted`.

Reply:
There are two kinds of `IsServerEncrypted`, one with header key `x-ms-server-encrypted` and `x-ms-request-server-encrypted`, they mean different things.

Reply:
and what's the difference?

Reply:
They are genuinely same at the moment, but see this:

>The Create File, Create Directory, Put Range, Set Directory Metadata, Set File Metadata, and Set File Properties operations now return the x-ms-request-server-encrypted response header. This header is set to true if the contents of the request have been successfully encrypted.

> The Get File, Get File Properties, and Get Directory Properties operations now return the x-ms-server-encrypted response header. This header is set to true if the file data and application metadata are completely encrypted. If the file is not encrypted, or if only parts of the file/application metadata are encrypted, this header is set to false.

 https://docs.microsoft.com/en-us/rest/api/storageservices/version-2017-04-17

Service deliberately made this difference for some reason.

Reply:
I don't think `RequestIsServerEncrypted ` is a good name. so either keep align with other SDK or escalate.

Reply:
How about `IsRequestServerEncrypted`?

Reply:
No, I don't see  any semantic difference between them

Reply:
I'll ask for Jeffrey's idea.

Reply:
This change is reverted to align with other SDK, however still pending for Jeffrey's input.